contrib/devtools/previous_release.sh

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env bash
+#
+# Copyright (c) 2018-2019 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+#
+# Build previous releases.
+
+export LC_ALL=C
+
+CONFIG_FLAGS=""
+FUNCTIONAL_TESTS=0
+DELETE_EXISTING=0
+USE_DEPENDS=0
+DOWNLOAD_BINARY=0
+CONFIG_FLAGS=""
+
+while getopts ":hfrdb" opt; do
+  case $opt in
+    h)
+      echo "Usage: .previous_release.sh [options] tag1 tag2"
+      echo "  options:"
+      echo "  -h   Print this message"
+      echo "  -f   Configure for functional tests"
+      echo "  -r   Remove existing directory"
+      echo "  -d   Use depends"
+      echo "  -b   Download release binary"
+      exit 0
+      ;;
+    f)
+      FUNCTIONAL_TESTS=1
+      CONFIG_FLAGS="$CONFIG_FLAGS --without-gui --disable-tests --disable-bench"
+      ;;
+    r)
+      DELETE_EXISTING=1
+      ;;
+    d)
+      USE_DEPENDS=1
+      ;;
+    b)
+      DOWNLOAD_BINARY=1
+      ;;
+    \?)
+      echo "Invalid option: -$OPTARG" >&2
+      exit 1
+      ;;
+  esac
+done
+
+shift $((OPTIND-1))
+
+if [ -z "$1" ]; then
+  echo "Specify release tag(s), e.g.: .previous_release v0.15.1"
+  exit 1
+fi
+
+if [ ! -d "build/releases" ]; then
+  echo "Create directory build/releases first"
+  exit 1
+fi
+
+if [ "$DOWNLOAD_BINARY" -eq "1" ]; then
+  HOST="${HOST:-$(./depends/config.guess)}"
+  case "$HOST" in
+    x86_64-*-linux*)
+      PLATFORM=x86_64-linux-gnu
+      ;;
+    x86_64-apple-darwin*)
+      PLATFORM=osx64
+      ;;
+    *)
+      echo "Not sure which binary to download for $HOST."
+      exit 1
+      ;;
+  esac
+fi
+
+pushd "$PWD/build/releases" || exit 1
+
+  for tag in "$@"
```

Comment:
nit: whitespace

Reply:
See also this discussion: https://github.com/mvdan/sh/issues/293

What do you think of the indenting with brackets suggestion?

pushd somewhere || exit 1
{
   do something
}
popd || exit 1

Reply:
@Sjors Brackets make it clear that the indentation is intentional :-)